src/app/features/tech-record/components/tech-record-amend-vin/tech-record-amend-vin.component.ts

```diff
@@ -42,10 +43,20 @@ export class AmendVinComponent {
     private store: Store<TechnicalRecordServiceState>
   ) {
     this.technicalRecordService.selectedVehicleTechRecord$
-      .pipe(take(1))
+      .pipe(take(1), takeUntil(this.destroy$))
```

Comment:
I had misunderstood the use there so I have gotten rid of the takeUntil in that instance. I may be wrong but the with the ngOnDestroy (I'd lost track of the changes I had made but I think it may be nessecary here) I think I still need it as with the subscription being on updateVinSuccess with take(1) it relies on us having a success. So if we didn't get a success then there may be an issue with a subscription being open? This would be the same for the subscription for the updateVrmSuccess too.

Reply:
Yes, it does rely on getting a success, so you are right. Good stop man. You can ignore this comment then.